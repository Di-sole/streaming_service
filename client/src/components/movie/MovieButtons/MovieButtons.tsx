import styles from './MovieButtons.module.scss';
import ColoredButton from '@/components/UI/buttons/ColoredButton/ColoredButton';
import Image from 'next/image';
import playIcon from '../../../../public/icons/movie/play.svg';
import bookmarkIcon from '../../../../public/icons/movie/bookmark.png';
import shareIcon from '../../../../public/icons//movie/share.png';
import { useTranslation } from 'next-i18next';
import { useRouter } from 'next/router';

const MovieButtons = () => {
    const { t } = useTranslation('movie');
    const { asPath } = useRouter();
    const router = useRouter();
    return (
        <div className={styles.container} role={'div-buttons'}>
            <ColoredButton
                onClick={() => router.push(`${asPath}?trailer`)}
                className={styles.container__item}
                color="lightGray"
                size="medium"
            >
                <Image src={playIcon} alt="playIcon" /> <span>{t('trailer')}</span>
            </ColoredButton>
            <ColoredButton className={styles.container__item} color="lightGray" size="medium">
                <Image src={bookmarkIcon} alt="bookmarkIcon" />
            </ColoredButton>
            <ColoredButton className={styles.container__item} color="lightGray" size="medium">
                <Image src={shareIcon} alt="shareIcon" />
            </ColoredButton>
        </div>
    );
};

export default MovieButtons;
