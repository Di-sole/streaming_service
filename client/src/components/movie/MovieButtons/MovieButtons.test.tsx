import { render, screen } from '@testing-library/react';
import MovieButtons from './MovieButtons';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

jest.mock('next/router', () => ({
    useRouter() {
        return {
            route: '/',
            asPath: '/',
        };
    },
}));

describe('MOVIEBUTTONS SNAPSHOTS TESTS', () => {
    test('MovieButtons should not be changed', () => {
        render(<MovieButtons />);

        const div = screen.getByTestId('buttons');
        expect(div).toMatchSnapshot();
    });
});
