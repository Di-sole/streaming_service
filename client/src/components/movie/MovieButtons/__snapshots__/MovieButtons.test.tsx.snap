// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MOVIEBUTTONS SNAPSHOTS TESTS MovieButtons should not be changed 1`] = `
<div
  class="container"
  data-testid="buttons"
>
  <button
    class="button button_medium button_lightGray container__item"
    type="button"
  >
    <img
      alt="playIcon"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
     
    <span>
      trailer
    </span>
  </button>
  <button
    class="button button_medium button_lightGray container__item"
    type="button"
  >
    <img
      alt="bookmarkIcon"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </button>
  <button
    class="button button_medium button_lightGray container__item"
    type="button"
  >
    <img
      alt="shareIcon"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </button>
</div>
`;
